lib/async_scheduler/cross_thread_usage_detector.rb

```diff
@@ -0,0 +1,18 @@
+module CrossThreadUsageDetector
+  class OriginalThreadAlreadySetError < StandardError; end
+  class CrossThreadUsageError < StandardError; end
+
+  def set_belonging_thread(original_thread_object_id)
+    if defined? @belonging_thread_object_id
+      raise OriginalThreadAlreadySetError.new("@belonging_thread_object_id is already set with #{@belonging_thread_object_id}, but it is attempted to set again with #{original_thread_object_id}.")
+    end
+
+    @belonging_thread_object_id = original_thread_object_id
+  end
+
+  def validate_used_in_original_thread!(another_thread_object_id)
```

Comment:
`validate_used_in_original_thread` looks like a method that compares with current thread.

I thought don't need arguments

Reply:
I actually implementing this, but changed it to pass `another_thread_object_id ` as an argument, because I did not want #validate_used_in_original_thread! to have side effects.

However, I have changed my mind. I think all the thread's object id related code should reside in Detector module.

Fixed in https://github.com/kudojp/async_scheduler/pull/29/commits/008bcf73ed9d75ac09e5e3445943478b2de0c4f1

Reply:
Also, I stopped passing arg to `#set_belonging_thread` as well.